source/src/AppEventLoop.cc

```diff
@@ -48,84 +48,72 @@ namespace dqm4hep {
     
     //-------------------------------------------------------------------------------------------------
     
-    void AppEventLoop::postEvent(AppEvent *pAppEvent)
-    {
-      if(nullptr == pAppEvent)
+    void AppEventLoop::postEvent(AppEvent *pAppEvent) {
+      if(nullptr == pAppEvent) {
         return;
-      std::lock_guard<std::recursive_mutex> lock(m_queueMutex);      
+      }
       // push event in the queue
-      m_eventQueue.push_front(std::shared_ptr<AppEvent>(pAppEvent));      
-      // sort the event queue by priority
-      std::sort(m_eventQueue.begin(), m_eventQueue.end(), [](AppEventPtr lhs, AppEventPtr rhs){
-        return lhs->priority() < rhs->priority();
-      });
+      std::lock_guard<std::recursive_mutex> lock(m_queueMutex);      
+      m_eventQueue.insert(pAppEvent);   
     }
     
     //-------------------------------------------------------------------------------------------------
     
-    void AppEventLoop::sendEvent(AppEvent *pAppEvent)
-    {
-      if(nullptr == pAppEvent)
+    void AppEventLoop::sendEvent(AppEvent *pAppEvent) {
+      if(nullptr == pAppEvent) {
         return;
-        
-      // take ownership of event ptr
-      std::unique_ptr<AppEvent> ptr(pAppEvent);
-      
+      }
       // process an event
       this->processEvent(pAppEvent);
+      delete pAppEvent;
     }
     
     //-------------------------------------------------------------------------------------------------
     
-    void AppEventLoop::clear()
-    {
+    void AppEventLoop::clear() {
       std::lock_guard<std::recursive_mutex> lock(m_queueMutex);
+      for(auto evt : m_eventQueue) {
+        delete evt;
+      }
       m_eventQueue.clear();
     }
     
     //-------------------------------------------------------------------------------------------------
 
-    int AppEventLoop::exec()
-    {
+    int AppEventLoop::exec() {
       m_running = true;
       m_quitFlag = false;
       
       m_timerThread = std::thread(&AppEventLoop::timerThread, this);
       
-      while(1)
-      {
-        if(m_quitFlag)
+      while(1) {
+        if(m_quitFlag) {
           break;
+        }
         
         // safely get the app event pointer 
-        AppEventPtr event;
+        AppEvent* event = nullptr;
         
         {
           std::lock_guard<std::recursive_mutex> lock(m_queueMutex);
           
-          if(!m_eventQueue.empty())
-          {
-            event = m_eventQueue.back();
-            m_eventQueue.pop_back();            
+          if(!m_eventQueue.empty()) {
+            event = *m_eventQueue.rbegin();
```

Comment:
I must be getting really tired but am not following here.
you say that if you have an event in the queue you take the last one...then delete it right away?

Reply:
Yes, exactly

Reply:
Mmm, okay. I didn't pay enough attention, I thought that you were deleting the event (not just erasing it from the queue). Thus why it didn't make sense.